diff --git a/nodejs/crypto.mjs b/nodejs/crypto.mjs
index 314f7fb2fdf084a98b7dd6d07fed8259467cf411..3e138a0538b82a731b8ca1e449314615e0a47fef 100644
--- a/nodejs/crypto.mjs
+++ b/nodejs/crypto.mjs
@@ -2,7 +2,7 @@
 // global scope. We import the generated JS file to make `wasm_bindgen`
 // available which we need to initialize our WASM code.
 import { parentPort } from 'node:worker_threads';
-import Comlink from 'comlink';
+import * as Comlink from 'comlink';
 import nodeEndpoint from 'comlink/dist/esm/node-adapter.mjs';
 import { CryptoUtils } from './crypto-wasm/index.js';
 
diff --git a/nodejs/index.mjs b/nodejs/index.mjs
index eae02e62c3b1268e377114ec88c967b48a75b2af..2d94b9a716f98ab50597c142f17752f00e80272f 100644
--- a/nodejs/index.mjs
+++ b/nodejs/index.mjs
@@ -1,5 +1,5 @@
 import { Worker } from 'node:worker_threads';
-import Comlink from 'comlink';
+import * as Comlink from 'comlink';
 import nodeEndpoint from 'comlink/dist/esm/node-adapter.min.mjs';
 import { Address, CryptoUtils, Transaction } from './main-wasm/index.js';
 import { clientFactory } from '../launcher/node/client-proxy.mjs';
diff --git a/nodejs/worker.mjs b/nodejs/worker.mjs
index e78edd6a31cc0bd495b956e27fe50b81495a3c38..2e9a633ae788782c66872e3730cdfd5daf0eabd1 100644
--- a/nodejs/worker.mjs
+++ b/nodejs/worker.mjs
@@ -2,7 +2,7 @@
 // global scope. We import the generated JS file to make `wasm_bindgen`
 // available which we need to initialize our WASM code.
 import { parentPort } from 'node:worker_threads';
-import Comlink from 'comlink';
+import * as Comlink from 'comlink';
 import nodeEndpoint from 'comlink/dist/esm/node-adapter.mjs';
 import websocket from 'websocket';
 import { Client } from './worker-wasm/index.js';
diff --git a/package.json b/package.json
index d3195f0f28c986e5130fccc441c3aaf77586ff4f..f87918f210e04204e03d4344920c8823f44cec0f 100644
--- a/package.json
+++ b/package.json
@@ -10,6 +10,7 @@
     "type": "git",
     "url": "https://github.com/nimiq/core-rs-albatross"
   },
+  "type": "module",
   "main": "./nodejs/index.js",
   "module": "./bundler/index.js",
   "types": "./types/bundler.d.ts",
@@ -24,6 +25,7 @@
     },
     "./web": {
       "browser": "./web/index.js",
+      "import": "./web/index.js",
       "types": "./types/web.d.ts"
     }
   },
@@ -49,7 +51,7 @@
     "proof-of-stake"
   ],
   "dependencies": {
-    "comlink": "^4.4.1",
+    "comlink": "^4.4.2",
     "websocket": "^1.0.34"
   }
 }
diff --git a/web/comlink.min.js b/web/comlink.min.js
deleted file mode 100644
index 8aa02e873c384d31dc7dc0108cdd966d220b30b6..0000000000000000000000000000000000000000
diff --git a/web/comlink.min.js.map b/web/comlink.min.js.map
deleted file mode 100644
index c6f4f26e0bebc5b2933fa924c8f897210df836ae..0000000000000000000000000000000000000000
diff --git a/web/comlink.min.mjs b/web/comlink.min.mjs
deleted file mode 100644
index cd008f0a0bab36f7cb5ca7da32fc36c17ff6f664..0000000000000000000000000000000000000000
diff --git a/web/comlink.min.mjs.map b/web/comlink.min.mjs.map
deleted file mode 100644
index 0c7ba803253e615a7a46553685d29ac1a3839982..0000000000000000000000000000000000000000
diff --git a/web/crypto.js b/web/crypto.js
index c93452a1dfa8b865c7f9900b1efad558a24480f3..c71ace68cd861c4829ac051d67235e12ad7aa3ab 100644
--- a/web/crypto.js
+++ b/web/crypto.js
@@ -2,7 +2,7 @@
 // global scope. We import the generated JS file to make `wasm_bindgen`
 // available which we need to initialize our WASM code.
 importScripts(
-    './comlink.min.js',
+    'comlink',
     './crypto-wasm/index.js',
 );
 
diff --git a/web/index.js b/web/index.js
index fc7904583a00fc3b3438cb370be03b37ee1078d7..4e4e99fa57ad5a4e11f9c4b10fbcda5084792bdd 100644
--- a/web/index.js
+++ b/web/index.js
@@ -1,4 +1,4 @@
-import * as Comlink from './comlink.min.mjs';
+import Comlink from 'comlink';
 import init, { Address, CryptoUtils, Transaction } from './main-wasm/index.js';
 import { clientFactory } from '../launcher/browser/client-proxy.mjs';
 import { cryptoUtilsWorkerFactory } from '../launcher/browser/cryptoutils-worker-proxy.mjs';
diff --git a/web/worker.js b/web/worker.js
index b8dc0a61fc34f0562aa1d9d775a930efeea8e795..4f02e1ff29468aca584604260b72176991c1487d 100644
--- a/web/worker.js
+++ b/web/worker.js
@@ -2,7 +2,7 @@
 // global scope. We import the generated JS file to make `wasm_bindgen`
 // available which we need to initialize our WASM code.
 importScripts(
-    './comlink.min.js',
+    'comlink',
     './worker-wasm/index.js',
 );
 
